src/Type/ContainerDynamicReturnTypeExtension.php

```diff
@@ -73,4 +75,18 @@ public function getTypeFromMethodCall(
 
         throw new ShouldNotHappenException();
     }
+
+    protected function getServiceId(Node $arg1): ?string
+    {
+        if ($arg1 instanceof String_) {
+            // @todo determine what these types are.
+            return $arg1->value;
+        }
+
+        if ($arg1 instanceof ClassConstFetch && $arg1->class instanceof FullyQualified) {
+            return (string) $arg1->class;
+        }
```

Comment:
I wonder if there's anyway to leverage PHPStan type system, which may abstract away some of this logic. But it may not be as accessible in these extensions

Reply:
I saw this issue #397 where you are adding a 'service-id-string' type which, correct me if I'm wrong, looks to make this plugin obsolete?

If so, I think this is a good starting point to get support for the described feature into phpstan-drupal, including tests that make following implementations think of the feature.

Also, this is my first contribution working with PHPStan related code, and the progress in #397 looks pretty advanced to me (for now).

Reply:
Do I understand correctly, that you would like to ask phpstan if the argument is of type 'string' or 'class-string' instead of this?

Do you have any idea where to start looking to come up with a solution that does that (because I am not really experienced in phpstan source)?

Reply:
Also I saw #397 which if I am not wrong, will make this plugin obsolete?

If so, in my opinion, this PR would make for good initial support of the feature. Any further (cleaner) solutions, for example in #397, using the type system would then also be checked against the tests for this feature.

Reply:
We can ignore that other PR, it's experimental. Let's work on this first